app/src/main/java/org/apache/fineract/ui/online/DashboardActivity.java

```diff
@@ -77,6 +80,10 @@ protected void onCreate(Bundle savedInstanceState) {
         }
     }
 
+    private void assignBaseFragment(FineractBaseFragment baseFragment) {
+        this.baseFragment = baseFragment;
+    }
+
```

Comment:
@EGOR-IND this looks very lengthy code. There is one solution if you want to try. Searchview assigned in every fragment is correct but instead of assigning fragment use `onPause()` in every fragment and in onPause method type `searchview.OnActionViewCollapsed()` it will make you work done with less code. For more reference follow this [pr made for similar issue](https://github.com/apache/fineract-cn-mobile/pull/149).

```suggestion
  ```

Reply:
Now here is the problem because my is actually a little bit shorter than appying on pause every fragment and also I think using interface is little better in software development sense because it keeps coupling low in a broad view.

Reply:
If you find it appropriate then it ok. 👍